sstables/kl/reader_impl.hh

```diff
@@ -295,13 +297,13 @@ private:
                 _deleted = (mask & column_mask::deletion) != column_mask::none;
                 _counter = false;
                 _state = state::CELL;
+                break;
             }
-            break;
         }
         case state::COUNTER_CELL:
```

Comment:
We should reorder the states here so that CELL comes after ATOM_MASK_2 and benefits from fallthrough behavior. Counters are relatively rare. Not for this series though, it's large enough.

Reply:
If I'm not mistaken, I moved the states so that state::CELL comes after this if/else in c41f49d2e543321d7044f0716b92c365d9988d26

Reply:
Yes, I saw later that the whole thing became moot.